docs/pages/versions/unversioned/sdk/firebase-recaptcha.md

```diff
@@ -0,0 +1,236 @@
+
+```tsx
+<FirebaseRecaptchaVerifierModal
+  ref={...}
+  firebaseConfig={...}
+  title='Prove you are human!'
+  cancelLabel='Close'
+/>
+```
+
+If you want a custom look & feel, then create your own `<Modal>` or display the `<FirebaseRecaptcha>` component inline in your screen.
+
+```tsx
+import { FirebaseRecaptchaVerifier } from 'expo-firebase-recaptcha';
+
+class CustomPhoneAuthScreen extends React.Component {
+  state = { 
+    recaptchaToken: ''
+  };
+
+  onPressSendVerificationCode = async () => {
+
+    // Create an application verifier from the reCAPTCHA token
+    const { recaptchaToken } = this.state;
+    if (!recaptchaToken) return;
+    const applicationVerifier = new FirebaseRecaptchaVerifier(recaptchaToken);
+
+    // Start phone autenthication
+    const phoneProvider = new firebase.auth.PhoneAuthProvider();
+    const verificationId = await phoneProvider.verifyPhoneNumber(
+      '+0123456789',
+      applicationVerifier
+    );
+  };
+
+  render() {
+    return (
+      <FirebaseRecaptchaVerifier
+        style={...}
+        firebaseConfig={...}
+
+        // Store the reCAPTCHA token when it has been verified
+        onVerify={recaptchaToken => this.setState({
+          recaptchaToken
+        })} />
+    );
+  }
+}
+```
+
+## API
+
+```js
+import {
+  FirebaseRecaptcha,
+  FirebaseRecaptchaVerifier,
+  FirebaseRecaptchaVerifierModal,
+  IFirebaseAuthApplicationVerifier
+} from 'expo-firebase-recaptcha';
+```
+
+### `<FirebaseRecaptchaVerifierModal>`
+
+Modal screen that is automatically shown and displays a reCAPTCHA widget. The ref to the component implements the `IFirebaseAuthApplicationVerifier` interface and can be used directly in the `verifyPhoneNumber` function.
+
+#### Props
+
+- **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
+- **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`. 
+- **title (string)** -- Title that is displayed on the top of the modal. The default is "reCAPTCHA".
+- **cancelLabel (string)** -- Label of the cancel button.
+
+
+### `<FirebaseRecaptcha>`
+
+The reCAPTCHA v3 widget displayed inside a web-view.
+
+#### Props
+
+- **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
+- **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`. 
+- **onVerify (function)** -- A callback that is invoked when reCAPTCHA has verified that the user is not a bot. The callback is provided with the reCAPTCHA token string. Example `onVerify={(recaptchaToken: string) => this.setState({recaptchaToken})}`.
+
+
+### `IFirebaseAuthApplicationVerifier`
+
+Interface describing a domain verification and abuse prevention verifier.
+
+```ts
+interface IFirebaseAuthApplicationVerifier {
```

Comment:
@ide - any suggestions for the naming of this type? i can't recall any existing examples of using the `I` for `Interface` prefix in our other sdk packages

Reply:
Just name it `FirebaseAuthApplicationVerifier` and put whatever makes the concrete type concrete in the class's name. Or if there is nothing that distinguishes the concrete type, we don't need the interface.

Reply:
I checked as well and couldn't find the `I` prefix or `Interface` suffix anywhere in the TS code of the packages. 

I've updated it to just `FirebaseAuthApplicationVerifier` 👍